Rubberduck.Core/UI/CodeExplorer/Commands/ImportCommand.cs

```diff
@@ -142,31 +179,116 @@ protected virtual ICollection<string> FilesToImport(object parameter)
 
         private void NotifyUserAboutAbortDueToUnsupportedFileExtensions(IEnumerable<string> fileNames)
         {
-            var firstUnsupportedFile = fileNames.First(filename => !_importableExtensions.Contains(Path.GetExtension(filename)));
+            var firstUnsupportedFile = fileNames.First(filename => !ImportableExtensions.Contains(Path.GetExtension(filename)));
             var unsupportedFileName = Path.GetFileName(firstUnsupportedFile);
             var message = string.Format(RubberduckUI.ImportCommand_UnsupportedFileExtensions, unsupportedFileName);
             MessageBox.NotifyWarn(message, DialogsTitle);
         }
 
         private void ImportFilesWithSuspension(ICollection<string> filesToImport, IVBProject targetProject)
         {
-            var suspensionResult = _parseManager.OnSuspendParser(this, new[] {ParserState.Ready}, () => ImportFiles(filesToImport, targetProject));
-            if (suspensionResult != SuspensionResult.Completed)
+            var suspendResult = _parseManager.OnSuspendParser(this, new[] {ParserState.Ready}, () => ImportFiles(filesToImport, targetProject));
+            var suspendOutcome = suspendResult.Outcome;
+            if (suspendOutcome != SuspensionOutcome.Completed)
             {
+                if (suspendOutcome == SuspensionOutcome.UnexpectedError || suspendOutcome == SuspensionOutcome.Canceled)
+                {
+                    ExceptionDispatchInfo.Capture(suspendResult.EncounteredException).Throw();
```

Comment:
this is really cool, but I had to look up the documentation for it. An explanatory comment along the lines of `// pretend the exception was thrown through this codepath instead of the parse manager` might be useful.